### PR TITLE
Also test optimized MIR

### LINKED ISSUE #256: Update comments in main re: when which optimizations are run
The main function says things like
```
    // we run the optimization passes inside miri
    // if we ran them twice we'd get funny failures due to borrowck ElaborateDrops only working on
    // unoptimized MIR
    // FIXME: add an after-mir-passes hook to rustc driver
    args.push("-Zmir-opt-level=0".to_owned());
```
I tried to find *where* these optimization passes are run, and couldn't. @eddyb when asked said the comment is outdated. However, if I remove the `args.push`, the test suite fails -- so something is clearly still going on.

I would appreciate if someone who knew what is happening could expand/update the comment -- or explain things to me and then I will update the comment. ;)

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #256

### CODE REVIEW

File: tests/compiletest.rs [hunk 1]
```diff
@@ -41,22 +44,32 @@ fn run_pass(path: &str) {
...
             // skip fullmir on nonhost
             return;
         }
         let sysroot = Path::new(&std::env::var("HOME").unwrap()).join(".xargo").join("HOST");
-        config.target_rustcflags = Some(format!("--sysroot {}", sysroot.to_str().unwrap()));
+        flags.push(format!("--sysroot {}", sysroot.to_str().unwrap()));
     }
+    if !opt {
+        flags.push("-Zmir-opt-level=0".to_owned());
+    }
```
Comment (RalfJung) [inline tests/compiletest.rs]: So we rely on the default being 2? Maybe it'd be better to always explicitly set the level.

**Comment by RalfJung** [comment]
What's with the code that's disabled in cast-rfc0401-vtable-kinds.rs?

File: tests/compiletest.rs (hunk 1 above)
Comment (oli-obk) [inline tests/compiletest.rs]: Good idea. Technically we should be testing all levels, but quadrupling our test times seems overkill

**Comment by oli-obk** [comment]
https://github.com/rust-lang/rust/issues/43358 (broken even for regular rust)

**Comment by RalfJung** [comment]
So it's the same as the other test? The comment referring to the Rust issue should then be added in both.
(Ideally we could disable the test only for optimizations turned on... Hm.)

**Comment by oli-obk** [comment]
Nope, the other test's issue is https://github.com/rust-lang/rust/issues/43359

> Ideally we could disable the test only for optimizations turned on... Hm.

I thought about that. Maybe I can convince compiletest to always disable optimizations on those tests.

**Comment by oli-obk** [comment]
> Maybe I can convince compiletest to always disable optimizations on those tests.

turns out I can. The tests are run twice (without optimizations) now, but oh well...

**Comment by RalfJung** [comment]
Nice! Looks good.

**Pull request merged**